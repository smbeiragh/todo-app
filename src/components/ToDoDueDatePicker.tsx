import * as React from 'react';
import classnames from 'classnames';
import DatePicker from 'react-datepicker';
import moment from 'moment';
import {Moment} from "moment";
import 'react-datepicker/dist/react-datepicker.css';

type Props = {
  value: Moment
  onChange: (value: Moment) => void
  className: string
}

type DefaultProps = {
  value: Moment
  onChange: (value: Moment) => void
  className: string
}

export class ToDoDueDatePicker extends React.Component<Props> {

  static defaultProps: DefaultProps = {
    value: moment(),
    onChange: (value: Moment) => {},
    className: ''
  }

  constructor(props: Props) {
    super(props)
  }

  render(){
    const { value, onChange, className } = this.props;
    return (
      <div className={classnames(className, 'c-toDoDueDatePicker')}>
        <DatePicker className={'c-toDoDueDatePicker__input'} selected={value} onChange={onChange} />
      </div>
    );
  }
}