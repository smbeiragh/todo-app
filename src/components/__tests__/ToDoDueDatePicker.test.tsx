import * as React from 'react';
import { ToDoDueDatePicker } from '../ToDoDueDatePicker';
import * as renderer from 'react-test-renderer';

test('ToDoDueDatePicker should render', () => {
  const component = renderer.create(
    <ToDoDueDatePicker />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
