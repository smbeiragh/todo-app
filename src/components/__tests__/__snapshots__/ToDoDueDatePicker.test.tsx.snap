// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoDueDatePicker should render 1`] = `
<div
  className="c-toDoDueDatePicker"
>
  <div>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <input
          className="c-toDoDueDatePicker__input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value="09/28/2018"
        />
      </div>
    </div>
  </div>
</div>
`;
